core/src/main/java/hudson/Functions.java

```diff
@@ -1302,6 +1304,24 @@ public static ThreadGroupMap sortThreadsAndGetGroupMap(ThreadInfo[] list) {
         return sorter;
     }
 
+    @Extension
+    public static class GlobalConfigHiddenByPermissionFilter extends DescriptorVisibilityFilter {
+
+        @SuppressWarnings("rawtypes")
+        @Override
+        public boolean filter(Object context, Descriptor descriptor) {
+
+            try {
+                if (descriptor instanceof GlobalConfiguration) {
+                    Jenkins.get().checkPermission(((GlobalConfiguration) descriptor).getPermission());
+                }
+            } catch (AccessDeniedException e) {
+                return false;
+            }
+            return true;
+        }
+    }
+
```

Comment:
this might be better in `GlobalConfiguration`? putting this in `Functions` seems a bit wild :D